spec/registry.md

```diff
@@ -1,11 +1,34 @@
 # MessageFormat 2.0 Default Function Registry
 
-This section describes the functions for which each implementation MUST provide
-a _function handler_ to be conformant with this specification.
-
-Implementations MAY implement additional _functions_ or additional _options_.
-In particular, implementations are encouraged to provide feedback on proposed
-_options_ and their values.
+This section defines the **standard** _functions_ which are REQUIRED for conformance with this specification,
+along with **optional** _functions_ that SHOULD be implemented to support
+additional functionality.
+
+Implementations MUST define and support each **standard** _function_ and
+MUST accept all _options_ and _option_ values defined as **standard** listed in this specification.
+
+When defined and supported by an implementation, each **optional** _function_ 
+MUST accept all _options_ and _option_ values listed as **standard** in this specification.
+
+Implementations SHOULD implement _options_ and _option_ values that are marked as **optional**.
+
+Implementations MAY implement _functions_ not defined in this specification.
+Such functions MUST use an implementation-defined _namespace_.
+In addition, implementations are encouraged to provide mechanisms for users to
+register and use user-defined _functions_ and their associated _functional handlers_.
```

Comment:
"are encouraged to" feels a bit too vague.

```suggestion
In addition, implementations SHOULD provide mechanisms for users to
register and use user-defined _functions_ and their associated _functional handlers_.
```

Reply:
SHOULD is probably ok.